IFC_oM/Enums/Discipline.cs

```diff
@@ -0,0 +1,48 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System.ComponentModel;
+
+namespace BH.oM.Adapters.IFC
+{
+    /***************************************************/
+
+    [Description("Enumerator allowing choosing to which discipline (and corresponding namespace) should IFC elements be converted on pull.")]
+    public enum Discipline
+    {
+        [Description("Default discipline to be used.")]
+        Undefined,
+        [Description("Elements to be converted to types from BH.oM.Environment. If no suitable conversion exists, default discipline to be used.")]
+        Environmental,
+        [Description("Elements to be converted to types from BH.oM.Structure. If no suitable conversion exists, default discipline to be used.")]
+        Structural,
+        [Description("Elements to be converted to types from BH.oM.Architecture. If no suitable conversion exists, default discipline to be used.")]
+        Architecture,
+        [Description("Elements to be converted to types from BH.oM.Physical. If no suitable conversion exists, default discipline to be used.")]
+        Physical,
+        [Description("Elements to be converted to types from BH.oM.Facade. If no suitable conversion exists, default discipline to be used.")]
+        Facade,
```

Comment:
Alphabetise these please.

Reply:
Not sure what is the point? `Undefined` needs to go first (default), others are likely to go out of sync at some point (`Facade` had just been added in Revit_Toolkit). I would keep it as is, to align with Revit_Toolkit.

Reply:
In previous discussions (links evade me at the moment, I think you were on the last discussion with Arnaud/Isak though when this reared up on BHoM a milestone or so back?) we agreed enums ought to be `Undefined` as first option, and subsequent options alphabetised unless a more logical order exists and makes sense (e.g. small -> smaller -> normal -> big -> bigger where an order is inherent and alphabetising would cause negative UX). This is because the order of the enum dictates the order on the UI for users. So where an enum is for public consumption, we should be consistent.
I'm happy for a justification to be put forward to have this enum be this order, but I'm afraid that "aligning with Revit_Toolkit" isn't strong enough on its own for me.

Reply:
Alphabetisation as a default does make sense to me, as in the long run is going to be most intuitive and navigable as we add more and more.
However I do think we need a link to any rules we want others to follow. 
Any compliance/style rules we want to enforce need to be written down and centrally available if we are expecting others to follow and comply. If that makes sense? Any change request (even small ones) make work for others (both authors and reviewers), so now is probably the time to start insisting on compliance and style changes only where they are documented so no ambiguity is left.
Can gradually build this central documentation up based on rules we need/want to enforce. Some will already have automated CI/CD checks, some will be, and perhaps will always remain, as a style guide. But all should be available in the Wiki
@FraserGreenroyd @pawelbaran does that make sense to both?

Reply:
Anything that gives me a lever to oppose @FraserGreenroyd's change requests makes sense to me 😉 Just to make sure we are on the same page, do you think it is worth alphabetising this enum, even if it can grow at some point in time? Then the order will break anyways (we cannot change the order of existing items because of serialisation issues I am afraid).

Reply:
Another thought on this topic: I am considering unifying certain aspects of Revit_Toolkit and IFC_Toolkit at some point, disciplines being one of the core items on my list. Therefore, I would rather keep this enum as is to make sure that the two toolkits are aligned, just in case. I think it is such a minor issue that we should not bother about it on the current, prototype stage, simply let it go. Would you be OK with that @FraserGreenroyd?

Reply:
I haven't seen any roadmap or user stories or documentation that highlights a unification requirement or desire, without it, I cannot agree to us doing something against practice in this _new_ toolkit. Ideally, we should have a reasonable set of user stories that define what the disciplines are that might be using this, and we could fill it in with all the options now, alphabetised, instead of adding a few now, and then adding more later. I'd see no harm in us doing a bit more planning on this to see what are the possible options we might want and bringing in the enum options now if serialisation is a concern.